mas-foundation/src/main/java/com/ca/mas/foundation/MAS.java

```diff
@@ -646,4 +647,21 @@ public static void registerMultiFactorAuthenticator(MASMultiFactorAuthenticator
     private static void unregisterMultiFactorAuthenticator(MASMultiFactorAuthenticator authenticator) {
         ConfigurationManager.getInstance().unregisterResponseInterceptor(authenticator);
     }
+
+    /**
+     * Uploads multipart form-data to server.
+     *
+     *  @param request The {@link MASRequest} to upload multipart form-data.
+     * @param multipart The multipart body  {@link MultiPart} .
+     * @param progressListener The  {@link MASProgressListener} to receive progress.
+     * @param callback The {@link MASCallback}.
+     */
+    public static void upload(MASRequest request, MultiPart multipart, MASProgressListener progressListener, MASCallback callback) throws MASException {
+        if(multipart == null || (multipart.getFilePart().isEmpty() && multipart.getFormPart()==null)){
```

Comment:
This line can throw NPE if multiplart.getFilePart is null. The condition should be like.
if(multipart == null || multipart.getFilePart() == null || multipart.getFormPart().isEmpty() ){

Reply:
getFilePart() will never be null.